*** interpreter/Python-3.6.9/Python/ceval_gil.h	2019-07-02 22:25:39.000000000 +0200
--- python_patches/ceval_gil.h	2019-07-25 21:35:28.447632153 +0200
***************
*** 79,87 ****
--- 79,94 ----
          Py_FatalError("PyMUTEX_UNLOCK(" #mut ") failed"); };
  
  #define COND_T PyCOND_T
+ 
+ #ifdef _3DS
+ #define COND_INIT(cond, mut) \
+     if (PyCOND_INIT(&(cond), &(mut))) { \
+         Py_FatalError("PyCOND_INIT(" #cond ") failed"); };
+ #else
  #define COND_INIT(cond) \
      if (PyCOND_INIT(&(cond))) { \
          Py_FatalError("PyCOND_INIT(" #cond ") failed"); };
+ #endif
  #define COND_FINI(cond) \
      if (PyCOND_FINI(&(cond))) { \
          Py_FatalError("PyCOND_FINI(" #cond ") failed"); };
***************
*** 138,146 ****
  #ifdef FORCE_SWITCHING
      MUTEX_INIT(switch_mutex);
  #endif
!     COND_INIT(gil_cond);
  #ifdef FORCE_SWITCHING
!     COND_INIT(switch_cond);
  #endif
      _Py_atomic_store_relaxed(&gil_last_holder, 0);
      _Py_ANNOTATE_RWLOCK_CREATE(&gil_locked);
--- 145,153 ----
  #ifdef FORCE_SWITCHING
      MUTEX_INIT(switch_mutex);
  #endif
!     COND_INIT(gil_cond, gil_mutex);
  #ifdef FORCE_SWITCHING
!     COND_INIT(switch_cond, switch_mutex);
  #endif
      _Py_atomic_store_relaxed(&gil_last_holder, 0);
      _Py_ANNOTATE_RWLOCK_CREATE(&gil_locked);
