*** interpreter/Python-3.6.9/pyconfig.h	2019-07-25 21:47:31.797630202 +0200
--- python_patches/pyconfig.h	2019-07-25 21:37:27.847631828 +0200
***************
*** 48,54 ****
  #define HAVE_ACOSH 1
  
  /* struct addrinfo (netdb.h) */
! #define HAVE_ADDRINFO 1
  
  /* Define to 1 if you have the `alarm' function. */
  /* #undef HAVE_ALARM */
--- 48,54 ----
  #define HAVE_ACOSH 1
  
  /* struct addrinfo (netdb.h) */
! #undef HAVE_ADDRINFO
  
  /* Define to 1 if you have the `alarm' function. */
  /* #undef HAVE_ALARM */
***************
*** 115,133 ****
  /* #undef HAVE_CHOWN */
  
  /* Define if you have the 'chroot' function. */
! #define HAVE_CHROOT 1
  
  /* Define to 1 if you have the `clock' function. */
  #define HAVE_CLOCK 1
  
  /* Define to 1 if you have the `clock_getres' function. */
! #define HAVE_CLOCK_GETRES 1
  
  /* Define to 1 if you have the `clock_gettime' function. */
! #define HAVE_CLOCK_GETTIME 1
  
  /* Define to 1 if you have the `clock_settime' function. */
! #define HAVE_CLOCK_SETTIME 1
  
  /* Define if the C compiler supports computed gotos. */
  /* #undef HAVE_COMPUTED_GOTOS */
--- 115,133 ----
  /* #undef HAVE_CHOWN */
  
  /* Define if you have the 'chroot' function. */
! #undef HAVE_CHROOT
  
  /* Define to 1 if you have the `clock' function. */
  #define HAVE_CLOCK 1
  
  /* Define to 1 if you have the `clock_getres' function. */
! #undef HAVE_CLOCK_GETRES
  
  /* Define to 1 if you have the `clock_gettime' function. */
! #undef HAVE_CLOCK_GETTIME
  
  /* Define to 1 if you have the `clock_settime' function. */
! #undef HAVE_CLOCK_SETTIME
  
  /* Define if the C compiler supports computed gotos. */
  /* #undef HAVE_COMPUTED_GOTOS */
***************
*** 228,234 ****
  
  /* Define to 1 if you have the declaration of `tzname', and to 0 if you don't.
     */
! #define HAVE_DECL_TZNAME 1
  
  /* Define to 1 if you have the device macros. */
  /* #undef HAVE_DEVICE_MACROS */
--- 228,234 ----
  
  /* Define to 1 if you have the declaration of `tzname', and to 0 if you don't.
     */
! #undef HAVE_DECL_TZNAME
  
  /* Define to 1 if you have the device macros. */
  /* #undef HAVE_DEVICE_MACROS */
***************
*** 295,301 ****
  /* #undef HAVE_FACCESSAT */
  
  /* Define if you have the 'fchdir' function. */
! #define HAVE_FCHDIR 1
  
  /* Define to 1 if you have the `fchmod' function. */
  #define HAVE_FCHMOD 1
--- 295,301 ----
  /* #undef HAVE_FACCESSAT */
  
  /* Define if you have the 'fchdir' function. */
! #undef HAVE_FCHDIR
  
  /* Define to 1 if you have the `fchmod' function. */
  #define HAVE_FCHMOD 1
***************
*** 313,319 ****
  #define HAVE_FCNTL_H 1
  
  /* Define if you have the 'fdatasync' function. */
! #define HAVE_FDATASYNC 1
  
  /* Define to 1 if you have the `fdopendir' function. */
  /* #undef HAVE_FDOPENDIR */
--- 313,319 ----
  #define HAVE_FCNTL_H 1
  
  /* Define if you have the 'fdatasync' function. */
! #undef HAVE_FDATASYNC
  
  /* Define to 1 if you have the `fdopendir' function. */
  /* #undef HAVE_FDOPENDIR */
***************
*** 792,798 ****
  /* #undef HAVE_SCHED_GET_PRIORITY_MAX */
  
  /* Define to 1 if you have the <sched.h> header file. */
! #define HAVE_SCHED_H 1
  
  /* Define to 1 if you have the `sched_rr_get_interval' function. */
  /* #undef HAVE_SCHED_RR_GET_INTERVAL */
--- 792,798 ----
  /* #undef HAVE_SCHED_GET_PRIORITY_MAX */
  
  /* Define to 1 if you have the <sched.h> header file. */
! #undef HAVE_SCHED_H
  
  /* Define to 1 if you have the `sched_rr_get_interval' function. */
  /* #undef HAVE_SCHED_RR_GET_INTERVAL */
***************
*** 834,840 ****
  /* #undef HAVE_SETGID */
  
  /* Define if you have the 'setgroups' function. */
! #define HAVE_SETGROUPS 1
  
  /* Define to 1 if you have the `sethostname' function. */
  /* #undef HAVE_SETHOSTNAME */
--- 834,840 ----
  /* #undef HAVE_SETGID */
  
  /* Define if you have the 'setgroups' function. */
! #undef HAVE_SETGROUPS
  
  /* Define to 1 if you have the `sethostname' function. */
  /* #undef HAVE_SETHOSTNAME */
***************
*** 994,1000 ****
  /* #undef HAVE_STRUCT_TM_TM_ZONE */
  
  /* Define if you have the 'symlink' function. */
! #define HAVE_SYMLINK 1
  
  /* Define to 1 if you have the `symlinkat' function. */
  /* #undef HAVE_SYMLINKAT */
--- 994,1000 ----
  /* #undef HAVE_STRUCT_TM_TM_ZONE */
  
  /* Define if you have the 'symlink' function. */
! #undef HAVE_SYMLINK
  
  /* Define to 1 if you have the `symlinkat' function. */
  /* #undef HAVE_SYMLINKAT */
***************
*** 1065,1071 ****
  /* #undef HAVE_SYS_RANDOM_H */
  
  /* Define to 1 if you have the <sys/resource.h> header file. */
! #define HAVE_SYS_RESOURCE_H 1
  
  /* Define to 1 if you have the <sys/select.h> header file. */
  #define HAVE_SYS_SELECT_H 1
--- 1065,1071 ----
  /* #undef HAVE_SYS_RANDOM_H */
  
  /* Define to 1 if you have the <sys/resource.h> header file. */
! #undef HAVE_SYS_RESOURCE_H
  
  /* Define to 1 if you have the <sys/select.h> header file. */
  #define HAVE_SYS_SELECT_H 1
***************
*** 1077,1083 ****
  #define HAVE_SYS_SOCKET_H 1
  
  /* Define to 1 if you have the <sys/statvfs.h> header file. */
! #define HAVE_SYS_STATVFS_H 1
  
  /* Define to 1 if you have the <sys/stat.h> header file. */
  #define HAVE_SYS_STAT_H 1
--- 1077,1083 ----
  #define HAVE_SYS_SOCKET_H 1
  
  /* Define to 1 if you have the <sys/statvfs.h> header file. */
! #undef HAVE_SYS_STATVFS_H
  
  /* Define to 1 if you have the <sys/stat.h> header file. */
  #define HAVE_SYS_STAT_H 1
***************
*** 1160,1166 ****
  
  /* Define to 1 if you don't have `tm_zone' but do have the external array
     `tzname'. */
! #define HAVE_TZNAME 1
  
  /* Define this if you have tcl and TCL_UTF_MAX==6 */
  /* #undef HAVE_UCS4_TCL */
--- 1160,1166 ----
  
  /* Define to 1 if you don't have `tm_zone' but do have the external array
     `tzname'. */
! #undef HAVE_TZNAME
  
  /* Define this if you have tcl and TCL_UTF_MAX==6 */
  /* #undef HAVE_UCS4_TCL */
***************
*** 1192,1198 ****
  #define HAVE_UTIMES 1
  
  /* Define to 1 if you have the <utime.h> header file. */
! #define HAVE_UTIME_H 1
  
  /* Define to 1 if you have the `wait3' function. */
  /* #undef HAVE_WAIT3 */
--- 1192,1198 ----
  #define HAVE_UTIMES 1
  
  /* Define to 1 if you have the <utime.h> header file. */
! #undef HAVE_UTIME_H
  
  /* Define to 1 if you have the `wait3' function. */
  /* #undef HAVE_WAIT3 */
