*** interpreter/Python-3.6.9/Python/thread.c	2019-07-02 22:25:39.000000000 +0200
--- python_patches/thread.c	2019-07-25 21:38:19.127631675 +0200
***************
*** 91,96 ****
--- 91,101 ----
  #include "thread_nt.h"
  #endif
  
+ #ifdef _3DS
+ #define PYTHREAD_NAME "ctr"
+ #include "thread_ctr.h"
+ #endif
+ 
  
  /*
  #ifdef FOOBAR_THREADS
***************
*** 382,388 ****
      3
  };
  
! static PyTypeObject ThreadInfoType;
  
  PyObject*
  PyThread_GetInfo(void)
--- 387,394 ----
      3
  };
  
! /* Conflicts with ThreadInfoType from libctru/include/3ds/synchronization.h. */
! static PyTypeObject _ThreadInfoType;
  
  PyObject*
  PyThread_GetInfo(void)
***************
*** 395,406 ****
      int len;
  #endif
  
!     if (ThreadInfoType.tp_name == 0) {
!         if (PyStructSequence_InitType2(&ThreadInfoType, &threadinfo_desc) < 0)
              return NULL;
      }
  
!     threadinfo = PyStructSequence_New(&ThreadInfoType);
      if (threadinfo == NULL)
          return NULL;
  
--- 401,412 ----
      int len;
  #endif
  
!     if (_ThreadInfoType.tp_name == 0) {
!         if (PyStructSequence_InitType2(&_ThreadInfoType, &threadinfo_desc) < 0)
              return NULL;
      }
  
!     threadinfo = PyStructSequence_New(&_ThreadInfoType);
      if (threadinfo == NULL)
          return NULL;
  
